src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -759,14 +803,17 @@ def create_disk(size_in_mb, cloud_properties, vm_cid = nil)
 
           logger.info("Created disk: #{disk.inspect}")
           raw_director_disk_cid = disk_pool.storage_list.any? ? DirectorDiskCID.encode(disk.cid, target_datastore_pattern: target_datastore_pattern) : disk.cid
+          @plugin_registry.run_post_hooks(__method__, self)
           # Return disk cid for the created disk.
           return raw_director_disk_cid
         end
+        # TODO: Run post hooks before raise?
```

Comment:
As a future improvement, you could either have the `run_post_hooks` method called in an ensure (to gain consistency of it always being called, even on exception), or add a separate hook specifically for when an error occurs in the CPI method (e.g. `run_error_hooks`)